app-office/texmacs/texmacs-1.99.12.recipe

```diff
@@ -0,0 +1,101 @@
+SUMMARY="A scientific editing platform designed to create beautiful technical documents"
+DESCRIPTION="TeXmacs provides a unified and user friendly framework for editing \
+structured documents with different types of content: text, mathematics, graphics, \
+interactive content, slides, etc.\
+Its rendering engine uses high-quality typesetting algorithms so as to produce \
+professionally looking documents, which can either be printed out or presented from \
+a laptop."
+HOMEPAGE="http://www.texmacs.org/"
+COPYRIGHT="Free Software Foundation"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgubi/texmacs/archive/texmacs-$portVersion-haiku-qt5.tar.gz"
+CHECKSUM_SHA256="607fa4db4676b78780f7af086feec2e2332dc18a371d79fa50d640de6db10b43"
+SOURCE_DIR="texmacs-texmacs-$portVersion-haiku-qt5/src"
+PATCHES="texmacs-$portVersion.patchset"
+ADDITIONAL_FILES="texmacs.rdef.in"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	texmacs$secondaryArchSuffix = $portVersion
+	app:TeXmacs
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libQt5Concurrent$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libguile$secondaryArchSuffix == 17.4.0
+	lib:libfreetype$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
+	lib:libltdl$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libiconv$secondaryArchSuffix
+	devel:libQt5Concurrent$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libguile$secondaryArchSuffix == 17.4.0
+	devel:libfreetype$secondaryArchSuffix
+	devel:libgmp$secondaryArchSuffix
+	devel:libltdl$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	cmd:guile_config$secondaryArchSuffix == 1.8.8
+	cmd:guile$secondaryArchSuffix == 1.8.8
```

Comment:
So far $secondaryArchSuffix isn't needed here:
```diff
-	cmd:guile_config$secondaryArchSuffix == 1.8.8
-	cmd:guile$secondaryArchSuffix == 1.8.8
+	cmd:guile_config == 1.8.8
+	cmd:guile == 1.8.8
```
(both use $prefix/bin so not needed)
On a sidenote, does TeXmacs need cmd:guild at runtime or buildtime?
cmd:guile_config should be provided by guile1_devel package (checking here now) so probably not needed at all here ...

Reply:
So far "cmd:guile_config" isn't needed here

Reply:
cmd:guild  is not needed. cmd:guile and cmd:guile_config are needed only at buildtime

Reply:
I put there  cmd:guile_config since I read that all dependencies have to be made explicit. However is ok for me to remove it. Is indeed provided by guile1_devel

Reply:
In the guile1 recipe they are declared as `cmd:guile_config$commandSuffix`and `cmd:guile$commandSuffix`

Reply:
I will wait to see if your build complete. Is not clear to me what to do here.

Reply:
> I will wait to see if your build complete. Is not clear to me what to do here.

Maybe move "cmd:guile  == 1.8.8" to BUILD_PREREQUIRES? (didn't check that one yet)

Reply:
> I put there cmd:guile_config since I read that all dependencies have to be made explicit. However is ok for me to remove it. Is indeed provided by guile1_devel

For packages relying on libSDL we don't need "cmd:sdl_config" also as it's provided by the _devel package (as long as pkgconfig file is supplied and finds it, which in this case it does)

Reply:
This is not clear to me, many packages seems to declare the depency explicitly: https://github.com/haikuports/haikuports/search?q=cmd%3Asdl_config&unscoped_q=cmd%3Asdl_config

Reply:
Also, what about the ARCHITECTURES variable? I put "?" on archs I cannot test myself, but if it works on yours then I can modify this variable accordingly?

Reply:
> Also, what about the ARCHITECTURES variable? I put "?" on archs I cannot test myself, but if it works on yours then I can modify this variable accordingly?

15 recipes using cmd:sdl_config, some don't use cmd:pkg_config some do, depending if they pick up sdl-config from the sdl_devel package I presume, but shouldn't be needed in most cases (a lot more recipes use the SDL libraries without the dependency) :)

Reply:
ok